code/modules/power/geothermal_power.dm

```diff
@@ -0,0 +1,117 @@
+/**
+ * Idea: a solar esqu controller that has a HE pipe looking thing running out from it which runs over Lava
+ * 			or any turf with special temp high enough.
+ *
+ *
+ *
+ */
+// Admin editable variable for power gen
+
+/obj/machinery/power/geothermal_controller
+	name = "Akureyri Geothermal Power Controller"
+	desc = "The Akureyri Geothermal Power Controller is the specialised geothermal Power solution offered to you by Frag Felix Storage."
+
+	icon = 'icons/obj/machines/power/geothermal.dmi'
+	icon_state = "controller_off"
+
+	dir = NORTH
+	anchored = TRUE
+	density = TRUE
+	use_power = USE_POWER_OFF
+	idle_power_usage = 0
+	active_power_usage = 0
+
+	var/range = 7
+	var/number_of_collectors = 0
+	var/list/collectors = list()
+	var/power_total = 0
+
+/obj/machinery/power/geothermal_controller/can_drain_energy(datum/actor, flags)
+	return FALSE
+
+/obj/machinery/power/geothermal_controller/process(delta_time)//TODO: remove/add this from machines to save on processing as needed ~Carn PRIORITY
+	if(machine_stat & BROKEN)
+		return
+	update_icon()
+	if(powernet)
+		if(use_power)
+			add_avail(power_total)
+
+
+/obj/machinery/power/geothermal_controller/update_icon_state()
+	switch(power_total)
+		if(50 to 500 KILOWATTS)
+			icon_state = "controller_low"
+		if(500 KILOWATTS to 1 MEGAWATTS)
+			icon_state = "controller_med"
+		if(1 MEGAWATTS to 2 MEGAWATTS)
+			icon_state = "controller_high"
+		if(2 MEGAWATTS to INFINITY)
+			icon_state = "controller_vhigh"
+		else
+			icon_state = "controller_idle"
+	if(use_power == USE_POWER_OFF)
+		icon_state = "controller_off"
+	return ..()
+
+
+/obj/machinery/power/geothermal_controller/Initialize(mapload)
+	. = ..()
+	for (var/obj/machinery/power/geothermal_collector/col in range(7, src))
```

Comment:
shouldnt this be using the controller's range var instead of hardcoded to be 7?

Reply:
made it dependant on a component